### PR TITLE
Helm chart testing

### LINKED ISSUE #535: Perform chart testing
The chart used and shipped could have more extensive testing. For this I was thinking of the [chart testing](https://github.com/helm/chart-testing) project. This would enable us to add more typical CI testing to the chart, too.

I've been, also, wondering if the chart as a separate repo would make this easier.

### PR DESCRIPTION
**Pull request by DirtyCajunRice**
fixes #535

Separated workflow to allow filtering trigger by file changes without needing multiple step workarounds

### CODE REVIEW

**Comment by tegioz** [comment]
Hi @scottrigby 👋 

Could you please review this PR when you have some time? You mentioned recently https://github.com/artifacthub/hub/issues/535#issuecomment-679219552 about using Chart Releaser as well, not sure if you'd like suggesting it as part of this or as a separate PR.

Thanks, much appreciated!

File: .github/workflows/helm.yml
Comment (scottrigby) [inline .github/workflows/helm.yml:18]: Are you adding this because you have separated `test` into its own job? Just a note that this isn't necessary if you keep these combined into a single `lint-test` job. Combined may also be a little faster as well, and less repeated step definitions.

**Review by scottrigby — CHANGES_REQUESTED** [review]
Thanks for the PR. This is something I planned to get to very soon. In general, have you seen https://github.com/helm/charts-repo-actions-demo?

File: .ct.yaml
Comment (DirtyCajunRice) [reply .ct.yaml:4]: Thats right. I was on autopilot by that point. (Other repos i use it in have dependencies within the same helm repo thus requiring it. Will fix)

File: .github/workflows/helm.yml
Comment (DirtyCajunRice) [reply .github/workflows/helm.yml:12]: Ah. I have seen varying opinions on whether or not a readme change justified a version bump. Yall are the boss and i have no opinion on it haha. I will fix.

File: .ct.yaml
Comment (DirtyCajunRice) [reply .ct.yaml:2]: If you have a moment to test yourself, maybe you will be able to come to another verdict than i was able to! I have found the chart-dirs option to be _**extremely**_ inconsistent and buggy when specifying anything other than `charts`. I attempted to use chart-dirs and point to the current directory, which failed catastrophically. I was not going to move the chart directory without the express approval of yall. Now that it is suggessted I can take care of that.

All of that said, if you know some secret about the `chart-dirs` function that all of my guys are oblivious to.. please drop some knowledge on me!

File: .github/workflows/helm.yml
Comment (DirtyCajunRice) [reply .github/workflows/helm.yml:3]: Ive seen unfortunate merges where master does not reflect the pr branch after merge. It was added as a "does no harm but can _potentially_ help". Ill remove it :)

File: .github/workflows/helm.yml
Comment (DirtyCajunRice) [reply .github/workflows/helm.yml:10]: That is exactly the purpose. As github is undoubtedly going to enforce minutes on actions sooner rather than later, this is just a safe default

File: .github/workflows/helm.yml
Comment (DirtyCajunRice) [reply .github/workflows/helm.yml:21]: I have done it a significantly more verbose way in the past. Good catch - I am going to have to go clean up a few workflows because of this

File: .github/workflows/helm.yml
Comment (DirtyCajunRice) [reply .github/workflows/helm.yml:41]: I am a huge proponent of pre-commit. Would you be opposed to me adding a minimal pre-commit config to the repo as well? (this would ensure the aforementioned is a non-issue, and could ensure things like sign offs are done)

File: .github/workflows/helm.yml
Comment (DirtyCajunRice) [reply .github/workflows/helm.yml:18]: I initially had this in the same workflow as the ci.yml file. Fixing :)

File: .github/workflows/helm.yml
Comment (scottrigby) [reply .github/workflows/helm.yml]: Love it. And if you feel motivated I'd support a PR to add this to the lint-test workflow in the Helm Actions demo repo 😉

File: .ct.yaml
Comment (scottrigby) [reply .ct.yaml]: I haven't seen an issue for that in the chart-testing repo, but would appreciate one if you can remember problems you've seen. As an example, I can say that this does work for the `stable` and `incubator` chart repos – this is the `ct` config file where those dirs are defined: https://github.com/helm/charts/blob/master/test/ct.yaml#L3-L5

File: .github/workflows/helm.yml
Comment (scottrigby) [reply .github/workflows/helm.yml]: Same here 😄 You can see we updated it recently https://github.com/helm/charts-repo-actions-demo/pull/25 Props to @davidkarlsen for initially pointing out the cleaner syntax in checkout action v2.

**Comment by scottrigby** [comment]
@DirtyCajunRice Looks like DCO is failing because you accepted my commit suggestion. This is a common problem on DCO-required repos. I made this browser plugin to help if you're interested: https://github.com/scottrigby/dco-gh-ui

You could also just paste your signoff line into the UI, but this helps automate that, make sure it's properly formatted etc.

File: charts/hub/ci/ct-values.yaml
Comment (scottrigby) [inline charts/hub/ci/ct-values.yaml:1]: Needs newline at EOF

Comment (scottrigby) [reply charts/hub/ci/ct-values.yaml:1]: Ah but also, I just said `hub` as an example, I see the chart name is `artifact-hub`. I personally think we should keep the chart directory name the same as the chart name for clarity to end users.

At risk of bikeshedding here, this brings up a question to me: should the chart name be `artifact-hub` or `artifacthub` to be consistent with the github org name? This is probably a question for @mattfarina, @tegioz and @cynthia-sg as they maintain this project.

**Review by scottrigby — CHANGES_REQUESTED** [review]

File: .github/workflows/helm.yml
Comment (scottrigby) [reply .github/workflows/helm.yml]: I think this is up to @mattfarina, @tegioz and @cynthia-sg as maintainers of this project. I don't have an opinion. 

Though this can also be caught in CI, and maybe it should? I haven't checked if the [GitHub Superlinter Action](https://github.com/github/super-linter) does newline at EOF checks, but may be worth looking into.

File: charts/hub/ci/ct-values.yaml
Comment (tegioz) [reply charts/hub/ci/ct-values.yaml]: We can rename it to `artifacthub`, sounds good to me 👍

File: .github/workflows/helm.yml
Comment (tegioz) [reply .github/workflows/helm.yml]: I think it's better if all this kind of checks are centralized in CI.

File: charts/hub/ci/ct-values.yaml
Comment (DirtyCajunRice) [reply charts/hub/ci/ct-values.yaml]: to condense replies: All above issues are resolved (i believe). I also fixed the secret to match the newer template of host for postgresql. I changed the name of the chart to artifacthub as decided, and fixed the commit issue. I leave the `Resolve conversation` buttons to the creators so i dont step on any toes. Lemme know if you see anything else!.

**Review by scottrigby — APPROVED** [review]
lgtm. Thanks for this 😄

File: cmd/hub/Dockerfile
Comment (tegioz) [inline cmd/hub/Dockerfile:21]: Could you please leave this one as `hub`? This is one of the cmd names, not the chart. Otherwise it'll fail building the containers when merged into master. Thanks!

Comment (DirtyCajunRice) [reply cmd/hub/Dockerfile:21]: @tegioz That was a mass `sed`  casualty. Fixing now.

**Comment by tegioz** [comment]
Regarding the **chart renaming**, there are some more things we may need to update (and I was wondering if we really should change it now as part of this PR):

- Chart `README.md`: artifact-hub is used there both for the repo and for the chart itself.
- Workflow `ci.yml`: artifact-hub is used there too for staging and production deployments.
- Charts index and packages in `docs/chart` directory: artifact-hub is used in all packages Chart.yaml files as well as in the index. All of them would need to be updated as well. Otherwise reindexing as is would list two different packages (with and without the dash).
- The chart with the dash is also deployed in staging and production, would need to change that as well.
- Once all of that is changed, in Artifact Hub, the package name `artifact-hub` will be deleted and a new one named `artifacthub` will be added. This means some entities will be deleted in cascade, like users subscriptions to that package, for example.

Maybe we can just step back and leave the chart renaming for a different PR if we really want to go ahead with it and think about all the implications carefully. I should have thought about all of this yesterday before agreeing to the name change, my bad.

**Comment by tegioz** [comment]
Other than that everything looks great! So if we go back to naming the chart `artifact-hub` we can go ahead and merge.

Thanks for your work @DirtyCajunRice 👍

**Comment by scottrigby** [comment]
Actually that is correct. What we would want to do before we name change is make a release where we mark the old name `deprecated` so it no longer appears in the hub. Then afterwards git move to the new name. But yes this should probably be done in a separate PR

**Comment by tegioz** [comment]
Hi @DirtyCajunRice 

Could you please revert the mass new line change commit? It has broken some tests in the frontend as some snapshots don't match anymore.

Thanks!

**Comment by DirtyCajunRice** [comment]
@tegioz oop went to sleep after and didnt wait for the tests. Yeah ill revert and exclude the ui

**Review by tegioz — APPROVED** [review]
LGTM 👍 Thanks!

**Pull request merged**
